tet.rs/src/tetris/board/my_board.rs

```diff
@@ -0,0 +1,497 @@
+        for i in 0..QUEUE_DISPLAY_SIZE {
+            if i >= max_len {
+                break;
+            }
+
+            let queue_at_index: OptionalTypeT = self.get_queue_at(i);
+
+            write!(
+                f,
+                "{}{}{}",
+                piece_color::get_piece_color(queue_at_index, false, false),
+                format!("{}", queue_at_index.unwrap()),
+                if i + 1 < QUEUE_DISPLAY_SIZE && i + 1 < max_len {
+                    " "
+                } else {
+                    "\r\n"
+                }
+            )?;
+        }
+
+        piece_color::reset_color(f)?;
+
+        for y in (0..(BOARD_HEIGHT + 4)).rev() {
+            write!(f, "{}", piece_representation::get_border())?;
+
+            for x in 0..BOARD_WIDTH {
+                let cell_state: bool = self.get_cell(x, y).unwrap();
+                if cell_state {
+                    write!(
+                        f,
+                        "{}{}",
+                        piece_color::get_piece_color(self.get_cell_type(x, y), false, false), // FIXME: color is hardcoded
+                        piece_representation::get_taken()
+                    )?;
+                } else {
+                    write!(f, "{}", piece_representation::get_empty())?;
+                }
+            }
+
+            piece_color::reset_color(f)?;
+            write!(f, "{}\r\n", piece_representation::get_border())?;
+        }
+
+        for _ in 0..BOARD_WIDTH + 2 {
+            write!(f, "{}", piece_representation::get_border())?;
+        }
+
+        write!(f, "\r\n")
+    }
+}
+
+impl Board {
+    pub fn new() -> Self {
+        let mut board: Self = Self {
+            board: BoardT::from_elem(BOARD_SIZE, false),
+            piece_types: [None; BOARD_SIZE],
+            active_piece_type: piece_type::PieceType::TI, // We'll set a new one right after
+            active_piece_position: piece_position::PiecePosition::new(
+                0,
+                0,
+                piece_rotation::PieceRotation::RN,
+            ), // Same, this will get a new value
+            queue: QueueT::new(),
+            second_bag: QueueT::new(),
+            held_piece: None,
+            has_held: false,
+            rng: WyRand::new(),
+        };
+
+        board.ensure_complete_queue(true);
+        board.spawn_next_piece();
+
+        board
+    }
+
+    pub fn get_cell(
+        &self,
+        x: piece_position::PositionT,
+        y: piece_position::PositionT,
+    ) -> Option<bool> {
+        match Board::xy_to_index(x, y) {
+            Ok(index) => Some(self.board[index]),
+            Err(_) => None,
+        }
+    }
+
+    pub fn set_cell(
+        &mut self,
+        x: piece_position::PositionT,
+        y: piece_position::PositionT,
+        val: bool,
+    ) {
+        match Board::xy_to_index(x, y) {
+            Ok(index) => self.board.set(index, val),
+            Err(_) => {}
```

Comment:
```suggestion
        if let Ok(index) = Board::xy_to_index(x, y) {
            self.board.set(index, val);
```

Reply:
I'm still a bit confused by this line, I'll have to think when reading through it later !
Still applying.